vms/platformvm/utxo/camino_locked_test.go

```diff
@@ -657,3 +662,704 @@ func TestVerifyLockUTXOs(t *testing.T) {
+					s.EXPECT().GetDepositOffer(testID).Return(&deposit.Offer{
+						UnlockHalfPeriodDuration: uint32((time.Minute).Seconds()),
+						Start:                    nowMinus10m,
+					}, nil)
+					s.EXPECT().LockedUTXOs(depositTxSet, gomock.Any(), locked.StateDeposited).Return(depositedUTXOs, nil)
+					return s
+				},
+				keys:         []*crypto.PrivateKeySECP256K1R{preFundedKeys[0]},
+				depositTxIDs: []ids.ID{testID},
+			},
+			want: []*avax.TransferableInput{
+				generateTestInFromUTXO(depositedUTXOs[0], sigIndices),
+			},
+			want1: []*avax.TransferableOutput{
+				generateTestOut(ctx.AVAXAssetID, depositedAmount, outputOwners, ids.Empty, ids.Empty),
+			},
+			want2: [][]*crypto.PrivateKeySECP256K1R{{preFundedKeys[0]}},
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			got, got1, got2, err := testHandler.UnlockDeposit(tt.args.state(ctrl), tt.args.keys, tt.args.depositTxIDs)
+			if tt.err != nil {
+				require.ErrorContains(t, err, tt.err.Error())
+				return
+			}
+
+			require.NoError(t, err)
+			require.Equal(t, tt.want, got, "Error asserting TransferableInputs: got = %v, want %v", got, tt.want)
+			require.Equal(t, tt.want1, got1, "Error asserting TransferableOutputs: got = %v, want %v", got1, tt.want2)
+			require.Equal(t, tt.want2, got2, "UnlockDeposit() got = %v, want %v", got2, tt.want2)
+		})
+	}
+}
+
+func TestVerifyUnlockDepositedUTXOs(t *testing.T) {
+	fx := &secp256k1fx.Fx{}
+	err := fx.InitializeVM(&secp256k1fx.TestVM{})
+	require.NoError(t, err)
+	err = fx.Bootstrapped()
+	require.NoError(t, err)
+	ctx := snow.DefaultContextTest()
+
+	testHandler := &handler{
+		ctx: ctx,
+		clk: &mockable.Clock{},
+		utxosReader: avax.NewUTXOState(
+			memdb.New(),
+			txs.Codec,
+		),
+		fx: fx,
+	}
+	tx := &dummyUnsignedTx{txs.BaseTx{}}
+	tx.Initialize([]byte{0})
+	var nilCreds *secp256k1fx.Credential
+	outputOwners, cred1 := generateOwnersAndSig(tx)
+	testID := ids.GenerateTestID()
+	sigIndices := []uint32{0}
+	utxoAmount := uint64(5)
+	depositedAmount := uint64(1000)
+	output := avax.TransferableOutput{
+		Asset: avax.Asset{ID: ctx.AVAXAssetID},
+		Out: &secp256k1fx.TransferOutput{
+			Amt:          depositedAmount,
+			OutputOwners: outputOwners,
+		},
+	}
+
+	now := time.Now()
+	nowMinus10m := uint64(now.Add(-10 * time.Minute).Unix())
+	type args struct {
+		chainState   func(ctrl *gomock.Controller) state.Chain
+		tx           txs.UnsignedTx
+		utxos        []*avax.UTXO
+		ins          []*avax.TransferableInput
+		outs         []*avax.TransferableOutput
+		creds        []verify.Verifiable
+		burnedAmount uint64
+		assetID      ids.ID
+	}
+	tests := map[string]struct {
+		args args
+		want map[ids.ID]uint64
+		err  error
+	}{
+		"Inputs Credentials Mismatch": {
+			args: args{
+				chainState: func(ctrl *gomock.Controller) state.Chain {
+					s := state.NewMockChain(ctrl)
+					return s
+				},
+				utxos: []*avax.UTXO{
```

Comment:
replace with helper func, plz - will be shorter and easier to read
look for other places like this below